src/test/it/bom.yml

```diff
@@ -0,0 +1,56 @@
+metadata:
+  # labels and annotations are key: value string attributes
+  labels:
+    name: bom-it
+    # Needed to make PCT happy
+    groupId: org.jenkins-ci.main
+    artifactId: jenkins-war
+    version: 2.118-artifact-manager-s3-SNAPSHOT
+spec:
+  core:
+    version: 2.118
+  plugins:
+    - groupId: "org.jenkins-ci.plugins.workflow"
+      artifactId: "workflow-api"
+      version:  2.28-rc333.0675e9e9cb4c
+    - groupId: "org.jenkins-ci.plugins"
+      artifactId: "copyartifact"
+      version: 1.40-rc298.60da56cfabea
+    # Maven HPI Plugin improperly resolves dependency trees, needs upper bounds check
+    - groupId: "org.jenkins-ci.plugins"
+      artifactId: "structs"
+      version: 1.14
+    - groupId: "org.jenkins-ci.plugins.workflow"
+      artifactId: "workflow-step-api"
+      version: 2.14
+    - groupId: "org.jenkins-ci.plugins.workflow"
+      artifactId: "workflow-job"
+      version: 2.20
+    - groupId: "org.jenkins-ci.plugins.workflow"
+      artifactId: "workflow-basic-steps"
+      version: 2.7
+    - groupId: "org.jenkins-ci.plugins.workflow"
+      artifactId: "workflow-support"
+      version: 2.18
+    - groupId: "org.jenkins-ci.plugins"
+      artifactId: "junit"
+      version: 1.24
+    - groupId: "org.jenkins-ci.plugins"
+      artifactId: "mailer"
+      version: 1.21
+    - groupId: "org.jenkins-ci.plugins"
+      artifactId: "scm-api"
+      version: 2.2.7
+    - groupId: "org.jenkins-ci.plugins"
+      artifactId: "script-security"
+      version: 1.44
+    - groupId: "org.jenkins-ci.plugins"
+      artifactId: "aws-java-sdk"
+      version: 1.11.264
+  environments:
+    - name: aws
+      plugins:
+        #TODO: BOM does not support FileSystem REFs, so somebody needs to add BOM auto-update logic here
```

Comment:
Uh, no, updating the BOM would not make sense here, because presumably you wish for the integration tests to be run as part of this plugin’s PR builds. So to verify some change in the plugin against these other tests, you would need to commit the change, then update the version here and commit again, before filing the PR! Does not make sense.

We could allow `${changelist}` to be substituted so you would have

```yaml
          version: 1.0-alpha-1${changelist}
```

but this would need to be updated after every MRP usage, which is still awkward. Filesystem path references could help, but it needs some exploration what they would look like. One option is

```yaml
          versionFrom: ../../../.flattened-pom.xml
```

assuming the local `process-resources` phase has been run. Alternately,

```yaml
          versionFrom: ../../..
```

where the version is inferred from the output of

```bash
mvn -Dset.changelist -Dexpression=project.version -Doutput=version help:evaluate
version=`cat version`
rm version
echo $version
```

Reply:
Yes, whatever works. I could easily do it in the custom WAR packager spec, but BOM would require a whatever standardized definition in https://github.com/jenkinsci/jep/pull/92